frontend/src/app/main/snap.cljs

```diff
@@ -132,13 +148,13 @@
 
         ;; Distance between the elements in an area, these are the snap
         ;; candidates to either side
-        lt-cand (inner-distance shapes-lt)
-        gt-cand (inner-distance shapes-gt)
+        lt-cand (inner-distance (mapv :selrect shapes-lt))
+        gt-cand (inner-distance (mapv :selrect shapes-gt))
 
         ;; Distance between the elements to either side and the current shape
         ;; this is the distance that will "snap"
-        lt-dist (mapv dist-lt shapes-lt)
-        gt-dist (mapv dist-gt shapes-gt)
+        lt-dist (->> (map dist-lt shapes-lt) (into #{}))
+        gt-dist (->> (map dist-gt shapes-gt) (into #{}))
```

Comment:
In this case `(into #{} (map dist-gt) shapes-gt)` will avoid additional seq creation (by `map`) thanks to transducers